core/domain/wipeout_service.py

```diff
@@ -315,57 +406,101 @@ def _delete_models(user_id, user_role, module_name):
+        snapshot_metadata_model_classes: list(class). The snapshot metadata
+            model classes that contain the activity IDs.
+        commit_log_model_class: class. The metadata model classes that
+            contains the activity IDs.
         commit_log_model_field_name: str. The name of the field holding the
-            activity id in the corresponding commit log model.
+            activity ID in the corresponding commit log model.
+
+    Returns:
+        (list(BaseSnapshotMetadataModel), list(BaseCommitLogEntryModel)).
+        The tuple of snapshot metadata and commit log models.
     """
     user_id = pending_deletion_request.user_id
-    metadata_models = snapshot_model_class.query(
-        snapshot_model_class.committer_id == user_id
-    ).fetch()
-    activity_ids = set([
-        model.get_unversioned_instance_id() for model in metadata_models])
+    snapshot_metadata_models = []
+    for snapshot_model_class in snapshot_metadata_model_classes:
+        snapshot_metadata_models.extend(
+            snapshot_model_class.query(ndb.OR(
+                snapshot_model_class.committer_id == user_id,
+                snapshot_model_class.commit_cmds_user_ids == user_id,
+                snapshot_model_class.content_user_ids == user_id,
+            )).fetch()
+        )
+    snapshot_metadata_ids = set(
+        model.get_unversioned_instance_id()
+        for model in snapshot_metadata_models
+    )
 
     commit_log_models = commit_log_model_class.query(
         commit_log_model_class.user_id == user_id
     ).fetch()
     commit_log_ids = set(
         getattr(model, commit_log_model_field_name)
-        for model in commit_log_models)
-    if activity_ids != commit_log_ids:
+        for model in commit_log_models
+    )
+    if snapshot_metadata_ids != commit_log_ids:
         logging.error(
             'The commit log and snapshot %s IDs differ. '
             'Snapshots without commit logs: %s, '
             'Commit logs without snapshots: %s.',
             activity_category,
-            list(activity_ids - commit_log_ids),
-            list(commit_log_ids - activity_ids))
+            list(snapshot_metadata_ids - commit_log_ids),
+            list(commit_log_ids - snapshot_metadata_ids))
 
-    activity_ids |= commit_log_ids
-    # The activity_mappings field might have only been partially generated, so
-    # we fill in the missing part for this activity category.
-    if activity_category not in pending_deletion_request.activity_mappings:
-        pending_deletion_request.activity_mappings[activity_category] = (
-            _generate_activity_to_pseudonymized_ids_mapping(activity_ids))
-        save_pending_deletion_requests([pending_deletion_request])
+    _save_activity_mappings(
+        pending_deletion_request,
+        activity_category,
+        snapshot_metadata_ids | commit_log_ids)
+
+    return (snapshot_metadata_models, commit_log_models)
+
+
+def _pseudonymize_activity_models(
```

Comment:
The reuse of the term "activity" is a bit tricky because "activity" historically referred to "exploration/collection". How are you planning to handle this transition?

Reply:
Yeah, let's go with "entity", but make sure to clarify in the docstring what counts as an entity. It's a bit too vague of a term but I can't think of an alternative.

Reply:
Done.